lua/neogit/status.lua

```diff
@@ -254,6 +256,10 @@ local function display_status()
   })
 
   status_buffer:set_lines(0, -1, false, output)
+  vim.api.nvim_buf_add_highlight(0, -1, 'NeogitBranch', 0, 5, 5 + status.head.branch:len() + 1)
+  if status.upstream ~= nil then
+    vim.api.nvim_buf_add_highlight(0, -1, 'NeogitRemote', 1, 5, 5 + status.upstream.branch:len() + 1)
+  end
```

Comment:
Why is this needed? Shouldn't the existing syntax file cover these cases?

Reply:
When I added the commit message header to the line, the highlighting extended to the full line, I only wanted to have the branch name highlighted as before.
I skimmed through vim documentation, but I couldn't figure out how to do that with Syntax regions, so I used the highlighting api instead.
There's probably a way by defining a Syntax region with a start like `^Head: \w` that resets the highlighting for the commit message, but then using the api sounds less involved to me